File: 2022-06-craft-conf-languages/examples/files.zig
const std = @import("std");

fn save_file(filename: []const u8) !void {
    const cwd = std.fs.cwd();
    var fp = try cwd.createFile(filename, .{});
    defer fp.close();
    _ = try fp.write("lots of interesting data");
}

pub fn main() !void {
    try save_file("out.dat");
}

File: 2022-06-craft-conf-languages/examples/files.go
package main

import "os"

func SaveFile(filename string) error {
	fp, err := os.Open(filename)
	if err != nil {
		return err
	}
	defer fp.Close()

	_, err = fp.WriteString("lots of interesting data")
	if err != nil {
		return err
	}

	return nil
}

func main() {
	SaveFile("out.dat")
}
